core/src/gateway_createfromjson.c

```diff
@@ -129,49 +131,72 @@ GATEWAY_HANDLE Gateway_CreateFromJson(const char* file_path)
     return gw;
 }
 
+static void rollbackModules(size_t success_modules_entries_count, GATEWAY_HANDLE gw, VECTOR_HANDLE modules_added_successfully)
```

Comment:
`success_modules_entries_count` is **always** the value of `VECTOR_size(modules_added_successfully)`. So why not call `VECTOR_size(modules_added_successfully)` _inside_ `rollbackModules()` instead of outside, and get rid of the unnecessary function parameter?